script/gn-check.js

```diff
@@ -0,0 +1,23 @@
+const cp = require('child_process')
+
+const { getOutDir } = require('./lib/utils')
+
+const OUT_DIR = getOutDir()
+if (!OUT_DIR) {
+  throw new Error(`No viable out dir: one of Debug, Testing, or Release must exist.`)
+}
+
+const gnCheckDirs = [
+  '//electron:electron_lib',
+  '//electron:electron_app',
+  '//electron:manifests',
+  '//electron/shell/common/api:mojo'
+]
+
+for (const dir of gnCheckDirs) {
+  const args = ['check', `../out/${OUT_DIR}`, dir]
+  const result = cp.spawnSync('gn', args, { stdio: 'inherit' })
```

Comment:
We need to execute the `gn` in src instead of global PATH to avoid wrong results when switching to old branches, can you reference `script/lint.js` to set the required environment values?

Reply:
We still need to pass `env` into the spawnSync options to pick up the new & improved path